### PR TITLE
fix: skeletize the weblink preview and only return the template if the URL has a value

### PR DESCRIPTION
**Pull request by CodeBaboon**
[DE42136](https://rally1.rallydev.com/#/289692574792d/custom/355050439968?detail=%2Fdefect%2F486176244948&fdp=true?fdp=true): Lessons > FACE Weblinks > Preview loading skeleton not appearing

NOTE: For the complete fix to the defect we will require the change to the `d2l-labs-attachment` component found in https://github.com/Brightspace/attachment/pull/24

Couple small fixes here:
- if the `entity` or `entity.link` values aren't set don't return any html
- skeletize the attachment component.

### CODE REVIEW

**Comment by CodeBaboon** [comment]
@ianlaidlaw So now I'm confused about what's happening with this file. I edited locally and had some strange spacing/tab diffs on the lines I edited (see [original diff](https://github.com/BrightspaceHypermediaComponents/activities/pull/1402/commits/80810c19c179c3e374c5da4649952847fbda2d30)). There were merge conflicts so I used the github editor to resolve them, keeping your portion of the file and just manually applying the affected code portions from my side, and now the whole file is showing as edited.

**Comment by ianlaidlaw** [comment]
> @ianlaidlaw So now I'm confused about what's happening with this file. I edited locally and had some strange spacing/tab diffs on the lines I edited (see [original diff](https://github.com/BrightspaceHypermediaComponents/activities/pull/1402/commits/80810c19c179c3e374c5da4649952847fbda2d30)). There were merge conflicts so I used the github editor to resolve them, keeping your portion of the file and just manually applying the affected code portions from my side, and now the whole file is showing as edited.

I checked out your branch, as it is now, and this file's line endings are set to `CRLF`, while the other files are set to `LF`. On master, this file is set to `LF`. I would think that your editor is just setting the line endings on save, but if you used the github editor, then I'm not entirely sure how this is happening.

**Review by ianlaidlaw — APPROVED** [review]
Code changes look good, though the file's line ending have been changed again. Not sure if that was deliberate or not

**Comment by CodeBaboon** [comment]
@ianlaidlaw It's not on purpose, but I also don't know why it's happening or how to prevent it.

**Review by ianlaidlaw — APPROVED** [review]

File: components/d2l-activity-editor/d2l-activity-content-editor/web-link/d2l-activity-content-web-link-url-preview.js [hunk 1]
```diff
@@ -39,20 +39,25 @@ class ContentWebLinkUrlPreview extends SkeletonMixin(LocalizeActivityEditorMixin
...
-			</d2l-labs-attachment>
+      <label class="d2l-label-text">${this.localize('content.previewLabel')}</label>
+      <div class="d2l-skeletize">
+        <d2l-labs-attachment
+          ?hidden="${this.skeleton}"
+          .attachmentId="${attachment.id}"
+          .attachment="${attachment}"
+        >
+        </d2l-labs-attachment>
+      </div>
```
Comment (ChrisKraljevicD2L) [inline components/d2l-activity-editor/d2l-activity-content-editor/web-link/d2l-activity-content-web-link-url-preview.js:60]: This whitespace has turned into spaces. I think it's supposed to be tabs?

image

File: components/d2l-activity-editor/d2l-activity-content-editor/web-link/d2l-activity-content-web-link-url-preview.js [hunk 2]
```diff
@@ -39,20 +39,25 @@ class ContentWebLinkUrlPreview extends SkeletonMixin(LocalizeActivityEditorMixin
...
+
 		return html`
-			<label class="d2l-label-text">${this.localize('content.previewLabel')}</label>
-			<d2l-labs-attachment
-				.attachmentId="${attachment.id}"
-				.attachment="${attachment}"
-			>
-			</d2l-labs-attachment>
+      <label class="d2l-label-text">${this.localize('content.previewLabel')}</label>
+      <div class="d2l-skeletize">
```
Comment (brissons) [inline components/d2l-activity-editor/d2l-activity-content-editor/web-link/d2l-activity-content-web-link-url-preview.js:53]: Just double checking, do we need the skeleton stuff here? It looks like there is some skeleton handling in the component itself: https://github.com/Brightspace/attachment/blob/7b6e48b5d24b607296cacc9aeb2db6733b410599/components/attachment.js#L331

Comment (CodeBaboon) [reply components/d2l-activity-editor/d2l-activity-content-editor/web-link/d2l-activity-content-web-link-url-preview.js:53]: I'll see if we can get away without the `d2l-skeletize` div.

Comment (CodeBaboon) [reply components/d2l-activity-editor/d2l-activity-content-editor/web-link/d2l-activity-content-web-link-url-preview.js]: Seems to have been unnecessary, removed in d10b9f1bd6bedeaaca46c965ce971668d7b27cc2

**Review by brissons — APPROVED** [review]

**Review by ianlaidlaw — APPROVED** [review]

**Pull request merged**